vm/src/builtins/object.rs

```diff
@@ -38,10 +38,26 @@ impl Constructor for PyBaseObject {
         // Ensure that all abstract methods are implemented before instantiating instance.
         if let Some(abs_methods) = cls.get_attr(identifier!(vm, __abstractmethods__)) {
             if let Some(unimplemented_abstract_method_count) = abs_methods.length_opt(vm) {
-                if unimplemented_abstract_method_count? > 0 {
-                    return Err(
-                        vm.new_type_error("You must implement the abstract methods".to_owned())
-                    );
+                let methods: Vec<PyStrRef> = abs_methods.try_to_value(vm)?;
+                let methods: Vec<String> = methods
+                    .into_iter()
+                    .map(|name| name.as_str().to_owned())
+                    .collect();
+                let methods = methods.join(", ");
```

Comment:
Originally I avoided allocating a vec by turning this into a tuple, then into iter, then joining with ", " using IterTools. 

```rust
let methods = abs_methods
.to_sequence(vm)
.tuple(vm)
.unwrap()
.into_iter()
.map(|s| {
    s.str(vm)
        .unwrap()
        .to_string()
})
.join(", ");
```

Afterwards, I saw in another part of the code base it was written this way. Is the vec one more preferrable?